src/test/kotlin/uk/gov/dluhc/printapi/service/IdFactoryTest.kt

```diff
@@ -0,0 +1,32 @@
+package uk.gov.dluhc.printapi.service
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+
+internal class IdFactoryTest {
+
+    private val idFactory = IdFactory()
+
+    @Test
+    fun `should generate requestId`() {
+        // Given
+
+        // When
+        val requestId = idFactory.requestId()
+
+        // Then
+        assertThat(requestId).hasSize(24)
```

Comment:
One minor suggestion, @NathanRussellValtech . 
We can assert the pattern as well.

<pre>
assertThat(requestId).matches(Regex("[\\da-f]{24}").toPattern())
 </pre>

Reply:
Yeah, I did consider something like that, but in the end I decided that we might be testing mongo's code rather than ours?
I think the thing that's important to us in respect of generating a request ID is that its 24 chars - that is the business rule that we need to fulfil. The format hasn't been defined as a business rule/requirement. We just need it to be 24 chars, and unique. Which I think the 2 tests in this class cover?

Reply:
I think the extra assertion is good to check that we are using mongo's code, rather than a random alphanumeric generator, but still, you can generate random values with the same hex format and they will most probably pass our uniqueness test. So not too worried about the current state of the test. Also, you pointed out the format hasn't been decided yet, so I am happy with it as well.